editor/plugins/camera_2d_editor_plugin.cpp

```diff
@@ -0,0 +1,145 @@
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "camera_2d_editor_plugin.h"
+
+#include "canvas_item_editor_plugin.h"
+#include "core/config/project_settings.h"
+#include "editor/editor_node.h"
+#include "editor/themes/editor_scale.h"
+#include "scene/gui/label.h"
+#include "scene/gui/menu_button.h"
+
+void Camera2DEditor::edit(Camera2D *p_camera) {
+	if (p_camera == selected_camera) {
+		return;
+	}
+	selected_camera = p_camera;
+}
+
+void Camera2DEditor::_menu_option(int p_option) {
+	switch (p_option) {
+		case MENU_SNAP_LIMITS_TO_VIEWPORT: {
+			if (selected_camera == nullptr) {
+				break;
+			}
+			selected_camera->set_limit(SIDE_LEFT, 0);
+			selected_camera->set_limit(SIDE_TOP, 0);
+			selected_camera->set_limit(SIDE_RIGHT, GLOBAL_GET("display/window/size/viewport_width"));
+			selected_camera->set_limit(SIDE_BOTTOM, GLOBAL_GET("display/window/size/viewport_height"));
+			break;
+		}
+		default:
+			break;
+	}
+}
+
+void Camera2DEditor::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_THEME_CHANGED: {
+			options->set_button_icon(get_editor_theme_icon(SNAME("Camera2D")));
+			options->get_popup()->set_item_icon(MENU_SNAP_LIMITS_TO_VIEWPORT, get_editor_theme_icon(SNAME("Camera2D")));
```

Comment:
Using two same icons is rather meh.
![image](https://github.com/user-attachments/assets/ad0f2f99-780b-4391-8a1d-5ec311c08c38)
Either remove the menu icon or pick a different one. Although these menus usually don't have icons, e.g.
![image](https://github.com/user-attachments/assets/bf88d182-d369-47a6-98d3-93c7b5ccbba5)